### PR TITLE
[#156755837] Add a Local development section to the README

### PR DESCRIPTION
**Pull request by fpersico**

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * ⚙️ [#156755837](https://www.pivotaltracker.com/story/show/156755837): Aggiungere la documentazione su come utilizzare un Proxy per lo sviluppo in locale

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: README.md [hunk 1]
```diff
@@ -173,6 +173,54 @@ $ bundle exec fastlane alpha
...
+
+Seguire la documentazione del repository [italia-backend](https://github.com/teamdigitale/italia-backend).
+
+#### WebView, HTTPS e ceriticati autofirmati
+
+Allo stato attuale react-native non consente di aprire WebView su url HTTPS con certificato autofirmato. L'IDP di test però fa utilizzo di HTTPS e di un certificato autofirmato. Per ovviare a questo problema è possibile installare in locale un Proxy che faccia da proxy-pass verso l'App Backend e l'IDP.
+
+##### Installazione di mitmproxy
+
+Un proxy semplice da utilizzare e addatto al nostro scopo è [mitmproxy](https://mitmproxy.org/). Per l'installazione seguire la [pagina di documentazione](https://docs.mitmproxy.org/stable/overview-installation/) del sito sito ufficiale.
```
Comment (cloudify) [inline README.md]: sito sito

File: README.md [hunk 2]
```diff
@@ -173,6 +173,54 @@ $ bundle exec fastlane alpha
...
+
+Una volta terminata l'installazione creare un file chiamato `mitmproxy_metro_bundler.py` con il seguente contenuto:
```
Comment (cloudify) [inline README.md]: mettiamo questo file nel repo dell'app, sotto la directory `contrib` (da creare), e in questa documentazione diciamo che il file si trova li

File: README.md [hunk 3]
```diff
@@ -173,6 +173,54 @@ $ bundle exec fastlane alpha
...
+
+```
+# This script is used during development to allow mitmproxy to intercept request on specific ports and make a proxy-pass on localhost
+# Usage: mitmweb --listen-port 9060 --web-port 9061 --ssl-insecure -s mitmproxy_metro_bundler.py
+from mitmproxy import ctx
+
+def request(flow):
+    if flow.request.host == "[SIMULATOR_HOST_IP]" and (flow.request.port == 8081 or flow.request.port == 8082 or flow.request.port == 8097):
```
Comment (cloudify) [inline README.md]: nello script, `SIMULATOR_HOST_IP` va preso da env var

File: README.md [hunk 4]
```diff
@@ -173,6 +173,54 @@ $ bundle exec fastlane alpha
...
+```
+
+Inserire al posto di `[SIMULATOR_HOST_IP]`:
+* `10.0.2.2` (Standard Android Emulator)
+* `10.0.3.2` (Genymotion Android Emulator)
+
+Avviare il Proxy con il seguente comando:
+
+```
+mitmweb --listen-port 9060 --web-port 9061 --ssl-insecure -s mitmproxy_metro_bundler.py
```
Comment (cloudify) [inline README.md]:
```
SIMULATOR_HOST_IP=XXXXX mitmweb --listen-port 9060 --web-port 9061 --ssl-insecure -s ./contrib/mitmproxy_metro_bundler.py
```

File: README.md [hunk 5]
```diff
@@ -173,6 +173,54 @@ $ bundle exec fastlane alpha
...
+```
+
+##### Installazione del certificato di mitmproxy all'interno dell'emulatore Android
+
+Installare il certifcato di mitmproxy all'interno dell'emulatore seguendo la [giuda](https://docs.mitmproxy.org/stable/concepts-certificates/) ufficiale.
```
Comment (cloudify) [inline README.md]: giuda -> guida :)

File: README.md [hunk 6]
```diff
@@ -175,6 +175,43 @@ $ bundle exec fastlane alpha
...
 
+### Sviluppo con App Backend e IDP di test in locale
+
+Per sviluppare l'applicazione utilizzando in locale l'App Backend e un IDP di test, è necessario seguire alcuni step aggiuntivi descritti di seguito.
+
+#### Installazione di App Backend e IDP di test
+
+Seguire la documentazione del repository [italia-backend](https://github.com/teamdigitale/italia-backend).
+
+#### WebView, HTTPS e ceriticati autofirmati
```
Comment (cloudify) [inline README.md]: typo

File: README.md [hunk 7]
```diff
@@ -175,6 +175,43 @@ $ bundle exec fastlane alpha
...
+
+##### Installazione di mitmproxy
+
+Un proxy semplice da utilizzare e addatto al nostro scopo è [mitmproxy](https://mitmproxy.org/). Per l'installazione seguire la [pagina di documentazione](https://docs.mitmproxy.org/stable/overview-installation/) del sito ufficiale.
+
+All'interno del repository è presente il file `scripts/mitmproxy_metro_bundler.py` che permette al proxy di intercettare le richieste verso il Simulatore e, solo in caso di porte specifiche, effettuare il proxy-pass verso localhost.
+Avviare il Proxy con il seguente comando:
+
+```
+SIMULATOR_HOST_IP=XXXXX mitmweb --listen-port 9060 --web-port 9061 --ssl-insecure -s scripts/mitmproxy_metro_bundler.py
```
Comment (cloudify) [inline README.md]: I think you have to provide the pat to the script relative to the project root (`./scripts/mitmproxy_metro_bundler .py`)

Comment (cloudify) [inline README.md]: I think you have to provide the pat to the script relative to the project root (`./scripts/mitmproxy_metro_bundler .py`)

File: README.md [hunk 8]
```diff
@@ -175,6 +175,43 @@ $ bundle exec fastlane alpha
...
+```
+
+Inserire al posto di `XXXXX`:
+* `10.0.2.2` (Standard Android Emulator)
+* `10.0.3.2` (Genymotion Android Emulator)
+
+##### Installazione del certificato di mitmproxy all'interno dell'emulatore Android
+
+Installare il certifcato di mitmproxy all'interno dell'emulatore seguendo la [giuda](https://docs.mitmproxy.org/stable/concepts-certificates/) ufficiale.
```
Comment (cloudify) [inline README.md]: typo

File: scripts/mitmproxy_metro_bundler.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# This script is used during development to allow mitmproxy to intercept request on specific ports and make a proxy-pass on localhost
```
Comment (cloudify) [inline scripts/mitmproxy_metro_bundler.py]: Needs shebang `#!/usr/bin/env python`

also needs `chmod a+x`

File: scripts/mitmproxy_metro_bundler.py [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+# Usage: SIMULATOR_HOST_IP=XXXXX mitmweb --listen-port 9060 --web-port 9061 --ssl-insecure -s mitmproxy_metro_bundler.py
+import os
+from mitmproxy import http
+
+def request(flow: http.HTTPFlow):
+    if flow.request.host == os.environ.get("SIMULATOR_HOST_IP") and (flow.request.port == 8081 or flow.request.port == 8082 or flow.request.port == 8097):
```
Comment (cloudify) [inline scripts/mitmproxy_metro_bundler.py]: indentation seems off

Comment (cloudify) [inline scripts/mitmproxy_metro_bundler.py]: where are these ports coming from?

File: README.md (hunk 7 above)
Comment (fpersico) [reply README.md]: If you are in the 'italia-app' project root you don't need it.

File: scripts/mitmproxy_metro_bundler.py (hunk 9 above)
Comment (fpersico) [reply scripts/mitmproxy_metro_bundler.py]: It is not an executable. It is a mitmproxy script. An example from the nonprofit repository https://github.com/mitmproxy/mitmproxy/blob/master/examples/simple/add_header.py

**Pull request merged**